parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -582,18 +595,31 @@ public EncodingStats convertEncodingStats(List<PageEncodingStats> stats) {
   }
 
   public static Statistics toParquetStatistics(
-      org.apache.parquet.column.statistics.Statistics stats) {
+    org.apache.parquet.column.statistics.Statistics stats) {
+    return toParquetStatistics(stats, ParquetProperties.DEFAULT_STATISTICS_TRUNCATE_LENGTH);
+  }
+
+  public static Statistics toParquetStatistics(
+      org.apache.parquet.column.statistics.Statistics stats, int truncateLength) {
     Statistics formatStats = new Statistics();
     // Don't write stats larger than the max size rather than truncating. The
     // rationale is that some engines may use the minimum value in the page as
     // the true minimum for aggregations and there is no way to mark that a
     // value has been truncated and is a lower bound and not in the page.
-    if (!stats.isEmpty() && stats.isSmallerThan(MAX_STATS_SIZE)) {
+    if (!stats.isEmpty() && withinLimit(stats, truncateLength)) {
       formatStats.setNull_count(stats.getNumNulls());
       if (stats.hasNonNullValue()) {
-        byte[] min = stats.getMinBytes();
-        byte[] max = stats.getMaxBytes();
-
+        byte[] min;
+        byte[] max;
+
+        if (stats instanceof BinaryStatistics && truncateLength != Integer.MAX_VALUE) {
```

Comment:
`truncateLength != MAX_VALUE` is special that the `truncator` handles anyway. I would not put it here.

Reply:
My concern here is if we don't put it here, there will be some objects created in getTruncator() and tuncateMin/Max() although it will bail out eventually. The default value is Integer.MAX_VALUE. So it would be the most of cases.